lib/browser/ws.ts

```diff
@@ -50,42 +53,48 @@ function make_signing_key(credentials: AWSCredentials, day: string, service_name
 
 function sign_url(method: string,
     url: URL,
-    credentials: AWSCredentials,
-    service_name: string,
+    signing_config: AwsSigningConfig,
     time: string = canonical_time(),
     day: string = canonical_day(time),
     payload: string = '') {
-
     const signed_headers = 'host';
+    const service = signing_config.service!;
     const canonical_headers = `host:${url.hostname.toLowerCase()}\n`;
     const payload_hash = Crypto.SHA256(payload, { asBytes: true });
     const canonical_params = url.search.replace(new RegExp('^\\?'), '');
     const canonical_request = `${method}\n${url.pathname}\n${canonical_params}\n${canonical_headers}\n${signed_headers}\n${payload_hash}`;
     const canonical_request_hash = Crypto.SHA256(canonical_request, { asBytes: true });
-    const signature_raw = `AWS4-HMAC-SHA256\n${time}\n${day}/${credentials.aws_region}/${service_name}/aws4_request\n${canonical_request_hash}`;
-    const signing_key = make_signing_key(credentials, day, service_name);
+    const signature_raw = `AWS4-HMAC-SHA256\n${time}\n${day}/${signing_config.credentials.aws_region}/${service}/aws4_request\n${canonical_request_hash}`;
```

Comment:
While we may not be able to remove region from credentials due to comaptibility, I think we should be using the signing configs region with a shim that sets it to the region in the credentials, if one exists.

Reply:
The signing_config is generated based on provider in `create_websocket_url()`. The `sign_url()` will simply use whatever passed in as parameter. We will generate a fresh signing_config every time we call `sign_url`.